### ISSUE TITLE
Multicast Autojoin

### ISSUE THREAD

**Issue by borismcfishtank** [braice/MuMuDVB#85]
I'm having an issue where the multicast is flooding due to cheapo switches, and I've tried to enable the multicast auto-join, but I keep getting an error of
 
> setsockopt IP_ADD_MEMBERSHIP ipv4 failed (multicast kernel?)'

I'm running Ubuntu 12.04 with kernel _3.13.0-43-generic_ 64Bit.

MumMuDVB is version 1.7.2_release, I've tried with the 1.6 and the 2.0.0 versions both from source and with the 1.6 version in the Ubuntu repos.

The read-me file has an asterisk referring to the auto_join but all it seems to be in reference too is relating to dodgy switches that propagate unregistered multicast.

Does anyone have any suggestions? I'm assuming there is something I'm missing with the kernel, ifconfig reports that multicast is enabled on the interface, and without the autojoin line in the config it all works fine, except that it evilly propagates multicast traffic everywhere.

**Comment by braice** [comment]
Hello

What is your switch model ? Maybe it is just not supporting multicast.



Brice


2015-01-26 0:04 GMT+01:00 borismcfishtank <notifications@github.com>:

**Comment by borismcfishtank** [comment]
The switch is an slm2005 linksys that has the option to not flood unregistered multicast, and I had it up and running and the network appeared fine and stable, but trying it at a different location with a second switch plugged in (as per original) if I stream any large amount of traffic ( for example and entire bouquet) it crashes the network completely. :(

I've had previous success with the slm series of linksys routers when multicasting with hardware encoders, and currently have running

As an aside, I've gone back to 14.04 and have tried mumudvb2 with the auto join command and it still does it.

Working as I'm typing, a wireshark from the TPLink shows both the hardware encoder stream and mumudvb propagating into it from the SLM, and from the SLM you initially only see what you registered for, but when you stop viewing a stream it continues to be streamed on that port :/ which is weird.

So this indeed does suggest that there is something untoward with the way it is handling multicast.

**Comment by braice** [comment]
Did you register a querier in your network ?

2015-01-30 21:33 GMT+01:00 borismcfishtank <notifications@github.com>:

**Comment by yo2loj** [comment]
Workaround is as follows:
- don't specify the multicast interface (option multicast_iface4 - this is the one creating the issue)
- add a route to your multicast interface as needed, like the one below:

`ip route add 224.0.0.0/4 dev eth0`

This sould solve the problem.

**Comment by m0yellow** [comment]
after seeing a slm20xx switch, I'll guess there isn't a querier enabled in the network.

**Comment by borismcfishtank** [comment]
Correct, and for the life of me I can't remember how I got around the issue!

I will investigate.

**Comment by m0yellow** [comment]
there is a igmp querier implemented in perl only somewhere on the net.
But beware, the port receiving the querier packets will get all multicasts.

**Comment by neidersun** [comment]
Hello. have the problem. how do I do that exactly? with me there is no muliticast interface
